common/src/main/java/org/jboss/da/common/version/VersionAnalyzer.java

```diff
@@ -17,62 +17,58 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-import lombok.Data;
-
 /**
- *
  * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
  */
 public class VersionAnalyzer {
 
+    private static final Pattern VERSION_PATTERN = Pattern
+            .compile("^" + VersionParser.RE_MMM + VersionParser.RE_QUALIFIER_WITH_SEPARATOR + "?");
+
     private final VersionParser versionParser;
+    private final List<String> suffixes = new ArrayList<>();
 
-    public VersionAnalyzer(VersionParser versionParser) {
-        this.versionParser = versionParser;
+    public VersionAnalyzer(List<String> suffixes) {
+        this.suffixes.addAll(suffixes);
+        this.versionParser = new VersionParser(suffixes);
     }
 
-    public VersionAnalysisResult analyseVersions(String querry, Collection<String> versions) {
+    public List<String> sortVersions(String querry, Collection<String> versions) {
         VersionComparator comparator = new VersionComparator(querry, versionParser);
-        List<String> sortedVersions = new ArrayList<>();
-        List<SuffixedVersion> parsedVersions = versions.stream()
-                .sorted(comparator)
-                .distinct()
-                .peek(v -> sortedVersions.add(v))
-                .map(versionParser::parse)
-                .collect(Collectors.toList());
-
-        SuffixedVersion version = versionParser.parse(querry);
-        Optional<String> bmv = findBiggestMatchingVersion(version, parsedVersions);
-
-        return new VersionAnalysisResult(bmv, sortedVersions);
+        List<String> sortedVersions = versions.stream().sorted(comparator).distinct().collect(Collectors.toList());
```

Comment:
Could be changed to versions.stream().distinct().sorted(comparator).collect(Collectors.toList()); maybe? (distinct before sorting)

Reply:
We actually don't expect that there will be duplicate versions, so no expected gain in sorting time if we first remove duplicates.
On the other hand having distinct after sorted makes the distinct calculation faster, as it doesn't need to construst and populate hash set.
(I didn't know this before your comment :D)

Reply:
ah ok. good then :-)